### ISSUE TITLE
Allow re-assembly of multi part models

### ISSUE THREAD

**Issue by randyre** [bambulab/BambuStudio#571]
When adding multipart models it asks to combine them into one model, but then you lose the multi-part multi-color aspects and then have to paint to color.  

Conversely, you can tell it no and keep it as a multipart, but then you have to assemble the parts like a kit after printing and some parts may not even be printable standalone.

Solution 1. would be to allow all the parts to remain separate, allow you to assign an AMS slot to the individual parts and then allow assembly

Solution 2. Do as Cura does and even though the parts are assembled in the correct placement, they remain separate and are allowed to be assigned to different extruders (or in this case AMS slots)

**Comment by RobinWhitfield** [comment]
If you split a model and move parts around, you can select everything and right click - "Assemble" to turn parts back into a multi-part model.

However I'm not sure why you would split the multi-part model in the first place, when you can move each part separately from object view.

Or am I fundamentally misunderstanding what you're trying to do?

**Comment by YifanWuBambu** [comment]
> If I accidentally move a part on a multipart model it's really hard to get it all aligned again. Please provide a way to re-assemble the objects once it's been split into parts and moved, just like when you first import the files.

Does undo (ctrl+z) help?

**Comment by randyre** [comment]
The problem is when you click on assemble it reassembles it into a solid object at the location they have been moved to. I would like it to reassemble it back just as if I had imported it initially as a multi-part model.

**Comment by randyre** [comment]
> If you split a model and move parts around, you can select everything and right click - "Assemble" to turn parts back into a multi-part model.
> 
> However I'm not sure why you would split the multi-part model in the first place, when you can move each part separately from object view.
> 
> Or am I fundamentally misunderstanding what you're trying to do?

I'm talking about both imported multipart models, as well as taking a single solid model and having the slicer turn it into multiple objects or parts.  This facilitates coloring the object way faster than painting it, but once it's broken into parts if you move any part, it won't reassemble, It will just turn it back to a solid with the parts in their last position.

As for musing control-Z if I take a single object and move it out of the group to say paint it I can no longer use Control-Z as it will not only move it back but also undo the paint.

Currently, other slicers can reassemble multi-part models after they've been imported. Why limit re-assembly only to initial import?

**Comment by randyre** [comment]
[deadpool-multicolour-remix-for-mmu-and-palette-model_files.zip](https://github.com/bambulab/BambuStudio/files/9972514/deadpool-multicolour-remix-for-mmu-and-palette-model_files.zip)

Import these files, then after it's imported move one of them out of the group to maybe paint it on a spot. Then try and get it reassembled for printing. If you try and hit center it drops everything to the build plate. you assemble it turns it back into a solid object.

**Comment by YifanWuBambu** [comment]
> [deadpool-multicolour-remix-for-mmu-and-palette-model_files.zip](https://github.com/bambulab/BambuStudio/files/9972514/deadpool-multicolour-remix-for-mmu-and-palette-model_files.zip)
> 
> Import these files, then after it's imported move one of them out of the group to maybe paint it on a spot. Then try and get it reassembled for printing. If you try and hit center it drops everything to the build plate. you assemble it turns it back into a solid object.

@randyre For this model, you do not need to move one of them out of group for painting.
Just specify the filament (color) for each part of the whole object in object list is enough.
![image](https://user-images.githubusercontent.com/110661856/200976498-dfcff0d9-d8e3-4485-9d68-7c09a0b5d587.png)

**Comment by YifanWuBambu** [comment]
If you want to choose each part more easily, please enter the Assembly View.
![image](https://user-images.githubusercontent.com/110661856/200976642-990fac1e-6bd9-4351-bda1-1206aa81a39d.png)

Here you can easily select each part of an object by simple click and paint(specify color) by numeric key.
It is also support to explode the object to make painting more easy.
![image](https://user-images.githubusercontent.com/110661856/200976824-ee9e2b1d-9e1f-4d11-a2c9-bd9f7f73aac2.png)
![image](https://user-images.githubusercontent.com/110661856/200976845-2a87d091-f4c4-44fc-a8f7-98b97daa5396.png)

**Comment by randyre** [comment]
I'll try that. If it does what I hope it does I'll close this ticket.

**Comment by randyre** [comment]
OK, close, but not 100% usable.  Need to be able to explode the object farther apart to make it useful. Also, you can only change the colors of each object, there is no paint allowed, so if say within a black object I want to change the color of the say the deapool logo on his chest I can only change all of the object not just a portion of it.
If we could explode an item in the assembly view, then go back to prepare to paint it, then back to the assembly view to re-assemble that would suffice.

**Comment by randyre** [comment]
Also, once the files are imported there's no way to assemble them properly unless I chose that in the initial import.

**Comment by randyre** [comment]
Still having this issue. Using center does not work either as the parts are different sizes. Would really like to have the ability to re-assemble parts after importing them or moving them.

**Comment by YifanWuBambu** [comment]
> 

Do you mean this?
![stack_objects](https://user-images.githubusercontent.com/110661856/203523803-c8d87ca1-d7e6-459f-b75b-d89562e491c8.gif)

**Comment by YifanWuBambu** [comment]
If it is still not what you want, could you help record a video for how other slicer handle this?

**Comment by randyre** [comment]
Yes, but select all but one object. Allow it to import and make one model with multi parts.  Now go back and pick the one part you didn't import and try and make it fit and line up where it's supposed to go. All it will do is import and drop it onto the build plate.

If you still need a video, I need a recommendation for software to record the screen with other than using my phone.

**Comment by randyre** [comment]
https://user-images.githubusercontent.com/60475095/203905738-33d8d419-0b7f-41fe-8944-99108b1a7df7.mp4

**Comment by YifanWuBambu** [comment]
@randyre 
Thanks for you explanation. I got what you mean. It has been added to our plan list.

**Comment by randyre** [comment]
@YifanWuBambu 

Any chance this is going to be assigned to someone and make it in the next build?

**Comment by YifanWuBambu** [comment]
> @YifanWuBambu
> 
> Any chance this is going to be assigned to someone and make it in the next build?

Hi @randyre ,
I added it to next month's schedule, but not in high priority.

**Comment by randyre** [comment]
@YifanWuBambu  June, and have had a couple of releases since the last post. Any chance this will be looked at for an upcoming release?

**Comment by bolsoncerrado** [comment]
We're finishing July already actually lol
